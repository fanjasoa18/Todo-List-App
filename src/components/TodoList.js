import React from "react";

function TodoList({setTodos, setDone, done = [], todos = []}) {

    const moveToDone = ({id}) => {
        const item = todos.find((todo) => todo.id === id);
        setDone([item, ...done]);

        const filterarray = todos.filter((todo) => todo.id !== id);
        setTodos(filterarray);
    }

    return(
        <div data-testid="todo-test" className="card">
            {todos.map((todo) => (
                <li data-testid="todo-list" className="list-item" key={todo.id}>
                    <input 
                        type="text" 
                        value={todo.title} 
                        className="list" 
                        onChange={(event) => event.preventDefault()} 
                    />
                    <div className="button">
                        <input type="checkbox" onChange={() => moveToDone(todo)} />   
                    </div>
                </li>
            ))
            }       
        </div>
    )
}

export default TodoList;