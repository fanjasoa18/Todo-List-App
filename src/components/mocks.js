export const todoListMocks = [
    {
        id: 1,
        title: 'banana'
    },
    {
        id: 2,
        title: 'pineapple'
    }
]