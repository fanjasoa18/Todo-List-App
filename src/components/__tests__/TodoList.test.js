import React from 'react';
import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import { create } from 'react-test-renderer';
import TodoList from '../TodoList';
import { todoListMocks as mocks } from '../mocks';
import "@testing-library/jest-dom/extend-expect";

describe('Todo', () => {
    afterEach(cleanup);

    it("renders correctly", () => {
        render(<TodoList todos={mocks} />);

        const todoElement = screen.getByTestId("todo-test");
        expect(todoElement).toBeInTheDocument();
    });


    test("snapshot renders correctly", () => {
        const renderer = create(<TodoList todos={mocks} />);
        const json = renderer.toJSON();

        expect(json).toMatchSnapshot();
    });


    it("maps todo item", () => {
        render(<TodoList todos={mocks} />);
        const todoElements = screen.getAllByTestId("todo-list");

        expect(todoElements.length).toEqual(mocks.length);

    });


    test("ckeckbox should move todolist", () => {
        const setDoneMock = jest.fn();
        const setTodoMock = jest.fn();

        render(<TodoList todos={mocks} setDone={setDoneMock} setTodos={setTodoMock} />);

        const todoElements = screen.getAllByTestId("todo-list");
  
        const [itemCheckbox] = screen.getAllByRole('checkbox');

        expect(itemCheckbox).toBeTruthy();

        const initialLength = mocks.length;

        fireEvent.click(itemCheckbox);

        const [todo1Mock] = mocks;

        const filteredMockArray = mocks.filter((todo) => todo.id !== todo1Mock.id);

        expect(todoElements.length).toEqual(mocks.length);
        expect(setTodoMock).toHaveBeenCalledWith(filteredMockArray);
        expect(filteredMockArray.length).toEqual(initialLength - 1)
    });

});