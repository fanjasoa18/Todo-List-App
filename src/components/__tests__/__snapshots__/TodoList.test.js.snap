// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo snapshot renders correctly 1`] = `
<div
  className="card"
  data-testid="todo-test"
>
  <li
    className="list-item"
    data-testid="todo-list"
  >
    <input
      className="list"
      onChange={[Function]}
      type="text"
      value="banana"
    />
    <div
      className="button"
    >
      <input
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </li>
  <li
    className="list-item"
    data-testid="todo-list"
  >
    <input
      className="list"
      onChange={[Function]}
      type="text"
      value="pineapple"
    />
    <div
      className="button"
    >
      <input
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </li>
</div>
`;